drivers/thermal/gov_step_wise.c

```diff
@@ -99,12 +99,15 @@ static void thermal_zone_trip_update(struct thermal_zone_device *tz, int trip_id
 	}
 
 	hyst_temp = trip_temp = trip.temperature;
-	if (tz->ops->get_trip_hyst) {
-		tz->ops->get_trip_hyst(tz, trip_id, &hyst_temp);
-		hyst_temp = trip_temp - hyst_temp;
-	}
 	trip_type = trip.type;
 
+	if (trip.hysteresis)
+		hyst_temp = trip_temp - trip.hysteresis;
+	else
```

Comment:
I don't think we need to warn about a zero hysteresis - what's wrong with this?:
```
    trip_temp = trip.temperature;
    hyst_temp = trip_temp - trip.hysteresis;
```

Reply:
Nothing really, the only reason for this was to make it more like the existing code in gov_bang_bang.c: https://github.com/torvalds/linux/blob/9b6de136b5f0158c60844f85286a593cb70fb364/drivers/thermal/gov_bang_bang.c#L22-L24

I'll make a simplified PR later.

Reply:
Updated now.